src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.Accessor.cs

```diff
@@ -247,48 +251,67 @@ private bool TryGetRowIdWorker(SqlConnection connection, TDatabaseId dataId, out
                 // not to the internal ROWID.
-                using (var resettableStatement = connection.GetResettableStatement(_select_rowid_from_0_where_1))
-                {
-                    var statement = resettableStatement.Statement;
+                using var resettableStatement = connection.GetResettableStatement(database == Database.WriteCache
+                    ? _select_rowid_from_writecache_table_where_0
+                    : _select_rowid_from_main_table_where_0);
 
-                    // Binding indices are 1-based.
-                    BindFirstParameter(statement, dataId);
+                var statement = resettableStatement.Statement;
 
-                    var stepResult = statement.Step();
-                    if (stepResult == Result.ROW)
-                    {
-                        rowId = statement.GetInt64At(columnIndex: 0);
-                        return true;
-                    }
+                // Binding indices are 1-based.
+                BindFirstParameter(statement, dataId);
+
+                var stepResult = statement.Step();
+                if (stepResult == Result.ROW)
+                {
+                    rowId = statement.GetInt64At(columnIndex: 0);
+                    return true;
                 }
 
                 rowId = -1;
                 return false;
             }
 
-            private void InsertOrReplaceBlob(
+            private void InsertOrReplaceBlobIntoWriteCache(
                 SqlConnection connection, TDatabaseId dataId,
                 byte[] checksumBytes, int checksumLength,
                 byte[] dataBytes, int dataLength)
             {
-                using var resettableStatement = connection.GetResettableStatement(_insert_or_replace_into_0_1_2_3_value);
-                var statement = resettableStatement.Statement;
+                using (var resettableStatement = connection.GetResettableStatement(_insert_or_replace_into_writecache_table_values_0_1_2))
+                {
+                    var statement = resettableStatement.Statement;
 
-                // Binding indices are 1 based.
-                BindFirstParameter(statement, dataId);
-                statement.BindBlobParameter(parameterIndex: 2, value: checksumBytes, length: checksumLength);
-                statement.BindBlobParameter(parameterIndex: 3, value: dataBytes, length: dataLength);
+                    // Binding indices are 1 based.
+                    BindFirstParameter(statement, dataId);
+                    statement.BindBlobParameter(parameterIndex: 2, value: checksumBytes, length: checksumLength);
+                    statement.BindBlobParameter(parameterIndex: 3, value: dataBytes, length: dataLength);
+
+                    statement.Step();
+                }
 
-                statement.Step();
+                // Let the storage system know it should flush this information
+                // to disk in the future.
+                Storage.EnqueueFlushTask();
```

Comment:
💡 Consider moving this all the way up to `SQLitePersistentStorage.PerformWriteAsync`

Reply:
Considered, but not finding that clear.